streams/src/main/java/org/apache/kafka/streams/processor/internals/AbstractTask.java

```diff
@@ -57,7 +57,8 @@ protected AbstractTask(TaskId id,
 
         // create the processor state manager
         try {
-            File stateFile = new File(config.getString(StreamsConfig.STATE_DIR_CONFIG), id.toString());
+            File jobStateDir = StreamThread.makeStateDir(jobId, config.getString(StreamsConfig.STATE_DIR_CONFIG));
```

Comment:
Kind of weird for this to be using `StreamThread` utilities. This almost seems like something that could go into `ClientUtils` if we can't simplify it to just `File.mkdirs()`.